man/systemd-creds.xml

```diff
@@ -0,0 +1,378 @@
+<?xml version='1.0'?> <!--*-nxml-*-->
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.5//EN"
+  "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd">
+<!-- SPDX-License-Identifier: LGPL-2.1-or-later -->
+
+<refentry id="systemd-creds"
+    xmlns:xi="http://www.w3.org/2001/XInclude">
+
+  <refentryinfo>
+    <title>systemd-creds</title>
+    <productname>systemd</productname>
+  </refentryinfo>
+
+  <refmeta>
+    <refentrytitle>systemd-creds</refentrytitle>
+    <manvolnum>1</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>systemd-creds</refname>
+    <refpurpose>Lists, shows, encrypts and decrypts service credentials</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <cmdsynopsis>
+      <command>systemd-creds</command>
+      <arg choice="opt" rep="repeat">OPTIONS</arg>
+    </cmdsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><command>systemd-creds</command> is a tool for listing, showing, encrypting and decrypting unit
+    credentials. Credentials are limited-size binary or textual objects that may be passed to unit
+    processes. They are primarily used for passing cryptographic keys (both public and private) or
+    certificates, user account information or identity information from the host to services.</para>
+
+    <para>Credentials are configured in unit files via the <varname>LoadCredential=</varname>,
+    <varname>SetCredential=</varname>, <varname>LoadCredentialEncrypted=</varname> and
+    <varname>SetCredentialEncrypted=</varname> settings, see
+    <citerefentry><refentrytitle>systemd.exec</refentrytitle><manvolnum>5</manvolnum></citerefentry> for
+    details.</para>
+  </refsect1>
+
+  <refsect1>
+    <title>Commands</title>
+
+    <para>The following commands are understood:</para>
+
+    <variablelist>
+      <varlistentry>
+        <term><command>list</command></term>
+
+        <listitem><para>Show a list of credentials passed into the current execution context. This command
+        shows the files in the directory referenced by the <varname>$CREDENTIALS_DIRECTORY</varname>
+        environment variable, and is intended to be executed from within service context.</para>
+
+        <para>Along with each credential name, the size and security state is shown. The latter is one of
+        <literal>secure</literal> (in case the credential is backed by unswappable memory,
+        i.e. <literal>ramfs</literal>), <literal>weak</literal> (in case it is backed by any other type of
+        memory), or <literal>insecure</literal> (if having any access mode that is not 0400, i.e. if readable
```

Comment:
Could this also show whether the key used is tpm, key or tpm+key?

Reply:
not really: this hows the keys passed to the app, i.e. the data *after* it ways acquired/decoded/decrypted, and at that point this information is not known anymore. 

And I think this is the right way: we should abstract away how we got the key and whether it was encrypted or not. i.e. this complexity is what systemd should implmenet and deal with, adn then once it acquired the key material it should just pass on the data in the simplest, most direct way we can: unencrypted, but backed by ramfs.

I mean, key here is: we want that apps can easily use the creds, even if we ourselve do a lot of work to acquire/decrypt them properly.

Reply:
(i mean, if we wanted we could add some xattr or so onto the creds files with some additional metadata about where we got the stuff, but I am not sure that'd be  good thing to let services know, i see it as a way of sandboxed isolation to hide this info away)

Reply:
makes sense, thanks